src/iface/ethernet.rs

```diff
@@ -389,19 +399,25 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
         match icmp_repr {
             // Respond to echo requests.
             Icmpv4Repr::EchoRequest { ident, seq_no, data } => {
-                let icmp_reply_repr = Icmpv4Repr::EchoReply {
-                    ident:  ident,
-                    seq_no: seq_no,
-                    data:   data
-                };
-                let ipv4_reply_repr = Ipv4Repr {
-                    src_addr:    ipv4_repr.dst_addr,
-                    dst_addr:    ipv4_repr.src_addr,
-                    protocol:    IpProtocol::Icmp,
-                    payload_len: icmp_reply_repr.buffer_len(),
-                    ttl:         64
-                };
-                Ok(Packet::Icmpv4(ipv4_reply_repr, icmp_reply_repr))
+                if ipv4_repr.dst_addr.is_unicast() {
```

Comment:
This code *asks* to be refactored into something like `icmpv4_reply` that would make the `Ipv4Repr` with the given `Icmpv4Repr`, and also have the condition for unicast addresses.

Reply:
100% agree